client/js/components/user/DpMasterToebList.vue

```diff
@@ -453,95 +450,37 @@ export default {
-      })
-
-      // Apply filters and search term
-      sortedList = sortedList.filter(item => this.searchedItems[item.ident] === 'in' && this.filteredItems[item.ident] === 'in')
-
-      this.rowItems = sortedList
-      this.$nextTick(() => {
-        if (this.isEditable) this.addCellIdsAndFields(this.dataTableElement.getElementsByTagName('tbody')[0], this.onPageItems)
-      })
-    },
-
-    /**
-     * Adjust the width of the inner element of the footer scrollbar to the width of the Table.
-     */
-    updateScrollbarWidth () {
-      this.scrollbar.firstChild.setAttribute('style', 'width:' + window.getComputedStyle(this.dataTableElement).width + ';height:1px;')
-    },
-
-    checkForUnreadChanges () {
-      // Check if there are unseen changes in the mastertöb list
-      dpApi.get(Routing.generate('DemosPlan_user_mastertoeblist_has_new_reportentry_ajax', { userId: this.userId }))
-        .then(data => {
-          if (data.code === 100 && data.success === true && data.hasNewReportEntry === true) {
-            // @todo find a more solid way to detect the target.
-            document.querySelector('.fa-bell').classList.add('color-status-failed-fill')
-          }
-        })
+    setSelectedItems (items) {
+      this.selectedItems = items
     },
 
-    addCellAttributes (row, item) {
-      row.forEach((cell, idx) => {
-        cell.setAttribute('data-ident', item.ident)
-        cell.setAttribute('data-field', this.headerFields[idx].field)
-      })
+    toggleAllCols (ev) {
+      Object.keys(this.filters).forEach(key => { this.filters[key] = ev.target.checked })
     },
 
     /**
-     * This method adds ids and fields to each cell as a data attribute. This is needed for later use in the event listener.
-     * The row index is set as a safety check to see later if there haven't been any shifts between cells and their
-     * corresponding items in this.onPageItems.
-     * @param table
-     * @param currentItems
+     * This method maps null and boolean values to their predefined display counterparts. Values in boolToStringFields need
+     * to be represented as 'x' (true)/ '' (false). Null values should be displayed as '-'
+     *
+     * @param value {string|boolean|null}
+     * @param field {string}
+     * @returns {string}
      */
-    addCellIdsAndFields (table, currentItems) {
-      const rows = Array.prototype.slice.call(table.getElementsByTagName('tr'))
-      rows.forEach((row, idx) => {
-        row.setAttribute('data-index', idx)
-        this.addCellAttributes(Array.prototype.slice.call(row.children), currentItems[idx])
-      })
+    transformValue (value, field) {
+      let returnVal = value
+      if (value === null || typeof value === 'undefined' || value === false) {
+        returnVal = '-'
+      }
+      if (this.boolToStringFields.includes(field)) {
+        returnVal = value ? 'x' : ''
+      }
+      return returnVal
```

Comment:
this can also be simplified:
```suggestion
      if (value === null || typeof value === 'undefined' || value === false) {
        return '-'
      }
      if (this.boolToStringFields.includes(field)) {
        return value ? 'x' : ''
      }
```
And one logical question:
Why does value be checked for `null`, `undefined` and `false`? Why can this variable have so many states?

Reply:
Done in [3ee5419](https://github.com/demos-europe/demosplan-core/pull/2621/commits/3ee541977317297a5722adfe38cd0dd1d04fbea0)
I don't know exactly why there are so many states, we get it like that from the backend. I guess it is `null` if it is not set in the backend and it can be `false` for boolean values. `undefined` maybe if something went wrong in the frontend?

Reply:
I realized we have to check for `boolToStringFields` first, because there should be displayed ' ' for `false` and otherwise there is '-' displayed. Done in [1b1fe82](https://github.com/demos-europe/demosplan-core/pull/2621/commits/1b1fe82b0f98518dfe1243d3bb039d040fa488e8)

Reply:
thank you for investigating and improving this !